pkg/gateway/api/api_grpc.go

```diff
@@ -1071,7 +1156,10 @@ func (s *grpcGatewayService) checkTrackRequest(
 	return envAPIKey, nil
 }
 
-func (s *grpcGatewayService) checkRequest(ctx context.Context) (*accountproto.EnvironmentAPIKey, error) {
+func (s *grpcGatewayService) checkRequest(
+	ctx context.Context,
+	role accountproto.APIKey_Role,
```

Comment:
Do we really need this role? Why we need this?

Reply:
Each role has different access to the data in the DB.
The client API key has read-only access to a single user's evaluations. The server API key has read-only access to all the flags and segment users in the environment. Because the server API key has access to sensitive data, we must apply different roles to the keys.

Unlike the client API Key, which can be exposed to the end user, the server API key must be managed on the server side, hiding it from the end user.

Reply:
Thanks! it's clear to me now :)